package.json

```diff
@@ -8,16 +8,39 @@
   "engines": {
     "node": ">=16.0"
   },
-  "main": "./index.js",
+  "type": "module",
+  "main": "./dist/cjs/index.js",
+  "module": "./dist/esm/index.js",
+  "types": "./dist/types/index.d.ts",
+  "typesVersions": {
```

Comment:
### typesVersions

The `typesVersions` needs to be more explicit in the current structure. Types are not resolving
with the wildcard in my test bed. Short term resolutions:

#### Explicit `typesVersions`

```json
{
  "typesVersions": {
    "*": {
      ".": [
        "types/index.d.ts"
      ],
      "types/app": [
        "types/types/app.d.ts"
      ],
      "types/app-client": [
        "types/types/app-client.d.ts"
      ],
      "types/app-spec": [
        "types/types/app-spec.d.ts"
      ],
      "types/transaction": [
        "types/types/transaction.d.ts"
      ]
    }
  }
}
```

Long term strategy would be adding a `./src/types/index.ts` entrypoint and treating `types` as a module (similar to `./src/testing`). It would require deduplication of `StateSpec` and refactoring. Maybe something to consider for `v2`

Reply:
Should be explicitly defined now